roofit/roofitcore/inc/RooMinimizer.h

```diff
@@ -20,147 +20,168 @@
+                    double n4 = 0.0, double n5 = 0.0, double n6 = 0.0, unsigned int npoints = 50);
+
+   void setProfile(bool flag = true) { _cfg.profile = flag; }
+   /// Enable or disable the logging of function evaluations to a RooDataSet.
+   /// \see RooMinimizer::getLogDataSet().
+   /// param[in] flag Boolean flag to disable or enable the functionality.
+   void setLoggingToDataSet(bool flag = true) { _loggingToDataSet = flag; }
+
+   /// If logging of function evaluations to a RooDataSet is enabled, returns a
+   /// pointer to a dataset with one row per evaluation of the RooAbsReal passed
+   /// to the minimizer. As columns, there are all floating parameters and the
+   /// values they had for that evaluation.
+   /// \see RooMinimizer::setLoggingToDataSet(bool).
+   RooDataSet *getLogDataSet() const { return _logDataSet.get(); }
+
+   static int getPrintLevel();
+
+   void setMinimizerType(std::string const &type);
+   std::string const &minimizerType() const { return _cfg.minimizerType; }
+
+   static void cleanup();
+   static RooFitResult *lastMinuitFit();
+   static RooFitResult *lastMinuitFit(const RooArgList &varList);
+
+   void saveStatus(const char *label, int status)
+   {
+      _statusHistory.push_back(std::pair<std::string, int>(label, status));
+   }
+
+   int evalCounter() const;
+   void zeroEvalCount();
+
+   ROOT::Fit::Fitter *fitter();
+   const ROOT::Fit::Fitter *fitter() const;
+
+   ROOT::Math::IMultiGenFunction *getMultiGenFcn() const;
+
+   int getNPar() const;
+
+   void applyCovarianceMatrix(TMatrixDSym const &V);
 
 private:
+   friend class RooAbsMinimizerFcn;
 
-  friend class RooAbsMinimizerFcn;
-
-  void profileStart() ;
-  void profileStop() ;
+   void profileStart();
+   void profileStop();
 
-  std::ofstream* logfile() ;
-  double& maxFCN() ;
+   std::ofstream *logfile();
+   double &maxFCN();
 
-  bool fitFcn() const;
+   bool fitFcn() const;
 
-  // constructor helper functions
-  void initMinimizerFirstPart();
-  void initMinimizerFcnDependentPart(double defaultErrorLevel);
+   // constructor helper functions
+   void initOldStyle(RooAbsReal &function);
+   void initNewStyle(std::shared_ptr<RooFit::TestStatistics::RooAbsL> &function);
+   void initMinimizerFirstPart();
+   void initMinimizerFcnDependentPart(double defaultErrorLevel);
+   void execSetters(); // Executes setters that set _fcn config as per given _cfg configuration
 
-  int _printLevel = 1;
-  int _status = -99;
-  bool _profile = false;
-  bool _loggingToDataSet = false;
-  bool _verbose = false;
-  bool _profileStart = false;
+   int _status = -99;
+   bool _profileStart = false;
+   bool _loggingToDataSet = false;
 
-  TStopwatch _timer;
-  TStopwatch _cumulTimer;
+   TStopwatch _timer;
+   TStopwatch _cumulTimer;
 
-  std::unique_ptr<TMatrixDSym> _extV;
+   std::unique_ptr<TMatrixDSym> _extV;
 
-  RooAbsMinimizerFcn *_fcn;
+   RooAbsMinimizerFcn *_fcn;
```

Comment:
```suggestion
   std::unique_ptr<RooAbsMinimizerFcn> _fcn;
```
Since you are already touching this line.

Then you also don't need to define a RooMinimizer constructor anymore!